src/main/java/eu/bbmri_eric/negotiator/governance/resource/ResourceServiceImpl.java

```diff
@@ -113,6 +132,48 @@ public List<ResourceWithStatusDTO> updateResourcesInANegotiation(
     return getResourceWithStatusDTOS(negotiationId);
   }
 
+  @Override
+  public ResourceResponseModel updateResourceById(Long id, ResourceCreateDTO resource) {
+    Resource res = repository.findById(id).orElseThrow(() -> new EntityNotFoundException(id));
+    res.setName(resource.getName());
+    res.setDescription(resource.getDescription());
+    Resource updatedResource = repository.save(res);
+    return modelMapper.map(updatedResource, ResourceResponseModel.class);
+  }
+
+  @Override
+  @Transactional
+  public List<ResourceResponseModel> addResources(List<ResourceCreateDTO> resourcesCreateDTO) {
+    ArrayList<Resource> resources = new ArrayList<Resource>();
+    DiscoveryService discoveryService =
+        discoveryServiceRepository
+            .findById(Long.valueOf("1"))
+            .orElseThrow(() -> new EntityNotFoundException("1"));
+    AccessForm accessForm =
+        accessFormRepository
+            .findById(Long.valueOf("1"))
```

Comment:
I'm not sure how correct it is to hardcode these values. If we keep it then please change the exceptions to meaningful error messages

Reply:
yes we have to decide, because when we add a resource by sync we have to set them. We should keep the 1 as default and arrange the exceptions as you suggested

Reply:
Fixed in 91dc15e17a50781c7de390647a7d33a5fdbed99a